cmake/mimalloc.cmake

```diff
@@ -0,0 +1,28 @@
+include(ExternalProject)
+
+set(MIMALLOC_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/mimalloc/install)
+
+ExternalProject_Add(
+  mimalloc
+  PREFIX mimalloc
+  GIT_REPOSITORY https://github.com/microsoft/mimalloc.git
+  GIT_TAG 81bd1b70b086d872bda515f99cfcf0d78f8ac86c # dev branch v1.6.3 + #240
+  INSTALL_DIR ${MIMALLOC_INSTALL_DIR}
+  # We need to patch mimalloc's default mmap ranges as they interfere with
+  # sanitizers
+  # https://github.com/llvm/llvm-project/blob/master/compiler-rt/lib/tsan/rtl/tsan_platform.h#L67-L72
+  PATCH_COMMAND
```

Comment:
Just to be sure add an `UPDATE_COMMAND ""` argument to this external project definition to make sure we don't try to update the checkout as it would re-trigger the patch command (as described in the PATCH_COMMAND section of [the documentation](externalproject_add_step))